File: python/knowledges/search_knowledge_base.py
import sys
import os
sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
from utils import MaiAgentHelper

API_KEY = '<your-api-key>'
KNOWLEDGE_BASE_ID = '<your-knowledge-base-id>'   # 你的知識庫 ID
SEARCH_QUERY = '高鐵'

# assert API_KEY != '<your-api-key>', 'Please set your API key'
# assert KNOWLEDGE_BASE_ID != '<your-knowledge-base-id>', 'Please set your knowledge base id'
# assert SEARCH_QUERY != '<your-search-query>', 'Please set your search query'


def main():
    """
    搜尋知識庫範例
    
    展示如何使用新的知識庫 API 搜尋知識庫內容
    """
    maiagent_helper = MaiAgentHelper(API_KEY)

    try:
        # 搜尋知識庫內容
        response = maiagent_helper.search_knowledge_base(KNOWLEDGE_BASE_ID, SEARCH_QUERY)
        
        if isinstance(response, list):
            search_results = response
            print(f"搜尋查詢：{SEARCH_QUERY}")
            print(f"找到 {len(search_results)} 個相關結果：")
            print("-" * 50)
            
            for idx, result in enumerate(search_results, 1):
                print(f"結果 {idx}:")
                print(f"內容: {result.get('text', '無內容')}")
                print(f"相似度分數: {result.get('score', 'N/A')}")
                print(f"檔案名稱: {result.get('chatbot_file', {}).get('filename', 'Unknown')}")
                print(f"頁碼: {result.get('page_number', 'N/A')}")
                print("-" * 50)
        else:
            print("搜尋結果格式不正確")
            
    except Exception as e:
        print(f"搜尋知識庫失敗：{e}")


if __name__ == '__main__':
    main() 

File: examples/csharp/knowledges/search_knowledge_base.cs
using System;
using System.Diagnostics;
using System.Text.Json;
using System.Threading.Tasks;
using Utils;

namespace MaiAgentExamples.Knowledges
{
    public static class SearchKnowledgeBase
    {
        public static string API_KEY = "<your-api-key>";
        public static string KNOWLEDGE_BASE_ID = "<your-knowledge-base-id>";   // 你的知識庫 ID
        public static string SEARCH_QUERY = "高鐵";

        public static async Task Main(string[] args)
        {
            // 搜尋知識庫範例
            //
            // 展示如何使用新的知識庫 API 搜尋知識庫內容

            Debug.Assert(API_KEY != "<your-api-key>", "Please set your API key");
            Debug.Assert(KNOWLEDGE_BASE_ID != "<your-knowledge-base-id>", "Please set your knowledge base id");
            Debug.Assert(SEARCH_QUERY != "<your-search-query>", "Please set your search query");

            var maiagentHelper = new MaiAgentHelper(API_KEY);

            try
            {
                // 搜尋知識庫內容
                var searchResponse = await maiagentHelper.search_knowledge_base(KNOWLEDGE_BASE_ID, SEARCH_QUERY);
                var response = JsonSerializer.Deserialize<JsonElement>(searchResponse.ToString()!);

                Console.WriteLine($"搜尋查詢：{SEARCH_QUERY}");
                Console.WriteLine(new string('-', 50));

                if (response.ValueKind == JsonValueKind.Array)
                {
                    var searchResults = response.EnumerateArray();
                    int idx = 0;
                    foreach (var result in searchResults)
                    {
                        idx++;
                        Console.WriteLine($"結果 {idx}:");
                        Console.WriteLine($"內容: {(result.TryGetProperty("text", out var text) ? text.GetString() : "無內容")}");
                        Console.WriteLine($"相似度分數: {(result.TryGetProperty("score", out var score) ? score.ToString() : "N/A")}");

                        if (result.TryGetProperty("chatbot_file", out var chatbotFile) &&
                            chatbotFile.TryGetProperty("filename", out var filename))
                        {
                            Console.WriteLine($"檔案名稱: {filename.GetString()}");
                        }
                        else
                        {
                            Console.WriteLine($"檔案名稱: Unknown");
                        }

                        Console.WriteLine($"頁碼: {(result.TryGetProperty("page_number", out var pageNum) ? pageNum.ToString() : "N/A")}");
                        Console.WriteLine(new string('-', 50));
                    }

                    Console.WriteLine($"找到 {idx} 個相關結果");
                }
                else
                {
                    Console.WriteLine("搜尋結果格式不正確");
                }

            }
            catch (Exception e)
            {
                Console.WriteLine($"搜尋知識庫失敗：{e}");
            }
        }
    }
}
